File: src/main/java/gay/skitbet/jackiro/managers/SetupManager.java
package gay.skitbet.jackiro.managers;

import gay.skitbet.jackiro.command.CommandContext;
import gay.skitbet.jackiro.utils.SetupSession;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class SetupManager {
    private static final Map<String, SetupSession> activeSetups = new ConcurrentHashMap<>();

    public static SetupSession startSetup(CommandContext context) {
        SetupSession setupSession = new SetupSession(context);
        activeSetups.put(context.getGuild().getId(), setupSession);
        return setupSession;
    }

    public static SetupSession getSetup(String guildId) {
        return activeSetups.get(guildId);
    }

    public static void endSetup(SetupSession setupSession) {
        activeSetups.remove(setupSession.getChannel().getGuild().getId());
    }
}

File: src/main/kotlin/gay/skitbet/jackiro/managers/SetupManager.kt
package gay.skitbet.jackiro.managers

import gay.skitbet.jackiro.command.CommandContext
import gay.skitbet.jackiro.utils.setup.SetupSession
import java.util.concurrent.ConcurrentHashMap

object SetupManager {
    private val activeSetups: MutableMap<String, SetupSession> = ConcurrentHashMap()

    fun startSetup(context: CommandContext): SetupSession {
        val setupSession = SetupSession(context)
        activeSetups[context.guild.id] = setupSession
        return setupSession
    }

    fun getSetup(guildId: String): SetupSession? {
        return activeSetups[guildId]
    }

    fun endSetup(setupSession: SetupSession) {
        activeSetups.remove(setupSession.channel.guild.id)
    }
}
